File: ch03/max.zig
const std = @import("std");
fn max(comptime T: type, x: T, y: T) T {
    if (x > y) {
        return x;
    } else {
        return y;
    }
}
pub fn main() !void {
    const x = 3;
    const y = 4;
    std.debug.print("The maximum of x and y is: {}\n", 
    .{max(i32, x, y)});
    const a = 3.14;
    const b = 2.71;
    std.debug.print("The maximum of a and b is: {}\n", 
    .{max(f64, a, b)});
}

File: ch03/max.cpp
#include <iostream>

// Define a function template for calculating the maximum of two values
template <typename T>
T max(T x, T y) {
    return (x > y) ? x : y;
}

int main() {
    // Declare and initialize two int variables
    int x = 3;
    int y = 4;
    // Call the max function template with int arguments
    std::cout << "The maximum of x and y is: " << max(x, y) << '\n';
    // Declare and initialize two double variables
    double a = 3.14;
    double b = 2.71;
    //  Call the max function template with double arguments
    std::cout << "The maximum of a and b is: " << max(a, b) << '\n';
    
    return 0;
}
